src/OWASPZAPDotNetAPI/OWASPZAPDotNetAPI/ClientApi.cs

```diff
@@ -143,7 +143,7 @@ private static Alert GetNewAlertFromAResponseSet(ApiResponseSet apiResponseSet)
                 Other = apiResponseSet.Dictionary.TryGetDictionaryString("other"),
                 Parameter = apiResponseSet.Dictionary.TryGetDictionaryString("param"),
                 Reference = apiResponseSet.Dictionary.TryGetDictionaryString("reference"),
-                Risk = string.IsNullOrWhiteSpace("") ?
+                Risk = string.IsNullOrWhiteSpace(apiResponseSet.Dictionary.TryGetDictionaryString("risk")) ?
                     Alert.RiskLevel.Low :

```

Comment:
I believe  the handling on the two lines below should be expanded to handle other risks not just low.

@gmaran23 ?

Reply:
@kingthorin the below two lines parses "risk" string from the output from the HTTP response, and would assign a particular Risk such as Low, Medium, High. However the line above (line # 146) was previously buggy in a way that always returned 'Low'. The suggested fix promptly parses allows for parsing the "risk" string and setting the appropriate Risk level.

Code changes looks ok to me for merging.

Reply:
Is this check needed though? The alert should always have a risk (and confidence).

Reply:
@thc202 I agree that the alert should always have a risk (and confidence). 

I think the reason the check was put in place for risk and confidence, rather than directly parsing has to do with the older version of zap where there were some deprecated risk or confidence (if I remember vaguely reliability was deprecated to promote the usage of confidence) values and the returned risk value wasn't matching the declared enums. 
So in order to avoid invalid casts, it was resorted that if a cast for confidence or risk fails,  a 'Low' could be returned.

Reply:
OK, makes sense, thank you.